src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -317,6 +320,51 @@ private async Task<Recommendation> GetSelectedRecommendationFromPreviousDeployme
             return selectedRecommendation;
         }
 
+        private async Task<Recommendation?> GetRecommendationForRedeployment(List<Recommendation> recommendations, CloudApplication deployedApplication, string deploymentProjectPath)
+        {
+            var targetRecipeId = !string.IsNullOrEmpty(deploymentProjectPath) ?
+                await GetDeploymentProjectRecipeId(deploymentProjectPath) : deployedApplication.RecipeId;
+
+            foreach (var recommendation in recommendations)
+            {
+                if (string.Equals(recommendation.Recipe.Id, targetRecipeId) && _deployedApplicationQueryer.IsCompatible(deployedApplication, recommendation))
+                {
+                    return recommendation;
+                }
+            }
+            return null;
+        }
+
+        private async Task<string> GetDeploymentProjectRecipeId(string deploymentProjectPath)
+        {
+            if (!_directoryManager.Exists(deploymentProjectPath))
+            {
+                throw new InvalidOperationException($"Invalid deployment project path. {deploymentProjectPath} does not exist on the file system.");
+            }
+
+            try
+            {
+                var recipeFiles = _directoryManager.GetFiles(deploymentProjectPath, "*.recipe");
+                if (recipeFiles.Length == 0)
+                {
+                    throw new InvalidOperationException($"Failed to find a recipe file at {deploymentProjectPath}");
+                }
+                if (recipeFiles.Length > 1)
+                {
+                    throw new InvalidOperationException($"Found more than one recipe file at {deploymentProjectPath}");

```

Comment:
It tells **what** but doesn't tell why there must not exist more than one recipe files. 

Something like this would make more sense
```
Found more than one recipe files at {path}. Only one recipe file per deployment project is supported.
```